### PR TITLE
fix `istioctl analyze` has the unexpected [IST0132] msg when analyzing the virtual service

### PR DESCRIPTION
**Pull request by monsteredc**
This PR is to fix issue #34653

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: galley/pkg/config/analysis/analyzers/testdata/virtualservice_host_not_found_gateway.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spec:
     istio: ingressgateway
   servers:
   - hosts:
-    - testing-01.com
+    - '*/testing-01.com'
```
Comment (howardjohn) [inline galley/pkg/config/analysis/analyzers/testdata/virtualservice_host_not_found_gateway.yaml:12]: shouldn't we have a test of all different forms to make sure they make sense?

Comment (monsteredc) [reply galley/pkg/config/analysis/analyzers/testdata/virtualservice_host_not_found_gateway.yaml]: @howardjohn Done. PTAL. Thanks!

**Comment by monsteredc** [comment]
@howardjohn PTAL. Thanks!

**Review by howardjohn — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**